components/brave_shields/browser/ad_block_service.cc

```diff
@@ -59,9 +60,48 @@ void AdBlockServiceDomainResolver(const char* host,
 
 }  // namespace
 
-std::string AdBlockService::g_ad_block_component_id_(kAdBlockComponentId);
-std::string AdBlockService::g_ad_block_component_base64_public_key_(
-    kAdBlockComponentBase64PublicKey);
+AdBlockService::SourceProviderObserver::SourceProviderObserver(
+    base::WeakPtr<AdBlockEngine> adblock_engine,
+    AdBlockSourceProvider* source_provider,
+    AdBlockResourceProvider* resource_provider,
+    scoped_refptr<base::SequencedTaskRunner> task_runner)
+    : adblock_engine_(adblock_engine),
+      source_provider_(source_provider),
+      resource_provider_(resource_provider),
+      task_runner_(task_runner) {
+  source_provider_->AddObserver(this);
+  source_provider_->LoadDAT(this);
+}
+
+AdBlockService::SourceProviderObserver::~SourceProviderObserver() {
+  source_provider_->RemoveObserver(this);
```

Comment:
ScopedObservation is much handier

Reply:
yep that would probably be better, but this is only a temporary "adapter" that won't be needed after the follow-up anyway